app/models/night.rb

```diff
@@ -1,10 +1,11 @@
 class Night < ApplicationRecord
   validates_presence_of :date
   belongs_to :couch
-  belongs_to :reservation
+  belongs_to :reservation, optional: true
 
   scope :all_for_couch, -> (couch) { where(couch: couch) }
   scope :between_check_in_check_out, -> (check_in, check_out) {
-    where(date: (Date.parse(check_in))..(Date.parse(check_out) - 1.day))
+    where(date: (Date.parse(check_in.to_s))..(Date.parse(check_out.to_s) - 1.day))
```

Comment:
This might be a bit unnecessary.  Basically it allows us to pass strings (e.g. "2017-02-22") OR Date objects (e.g. `Date.current`) to the scope.  I was thinking about programming for least surprise, and it just felt right that this scope should be able to handle Date objects.